services/sqs/src/main/java/software/amazon/awssdk/services/sqs/batchmanager/BatchOverrideConfiguration.java

```diff
@@ -0,0 +1,449 @@
+ * are not specified.
+ */
+@SdkPublicApi
+public final class BatchOverrideConfiguration implements ToCopyableBuilder<BatchOverrideConfiguration.Builder,
+    BatchOverrideConfiguration> {
+
+    private final Integer maxBatchItems;
+    private final Integer maxBatchKeys;
+    private final Integer maxBufferSize;
+    private final Duration maxBatchOpenDuration;
+    private final Duration visibilityTimeout;
+    private final Duration longPollWaitTimeout;
+    private final Duration minReceiveWaitTime;
+    private final Integer maxDoneReceiveBatches;
+    private final List<MessageSystemAttributeName> messageSystemAttributeNames;
+    private final List<String> receiveMessageAttributeNames;
+    private final Boolean adaptivePrefetching;
+    private final Integer maxInflightReceiveBatches;
+
+    public BatchOverrideConfiguration(Builder builder) {
+        this.maxBatchItems = Validate.isPositiveOrNull(builder.maxBatchItems, "maxBatchItems");
+        this.maxBatchKeys = Validate.isPositiveOrNull(builder.maxBatchKeys, "maxBatchKeys");
+        this.maxBufferSize = Validate.isPositiveOrNull(builder.maxBufferSize, "maxBufferSize");
+        this.maxBatchOpenDuration = Validate.isPositiveOrNull(builder.maxBatchOpenDuration, "maxBatchOpenDuration");
+        this.visibilityTimeout = Validate.isPositiveOrNull(builder.visibilityTimeout, "visibilityTimeout");
+        this.longPollWaitTimeout = Validate.isPositiveOrNull(builder.longPollWaitTimeout, "longPollWaitTimeout");
+        this.minReceiveWaitTime = Validate.isPositiveOrNull(builder.minReceiveWaitTime, "minReceiveWaitTime");
+        this.messageSystemAttributeNames = builder.messageSystemAttributeNames;
+        this.receiveMessageAttributeNames = builder.receiveMessageAttributeNames;
+        this.adaptivePrefetching = builder.adaptivePrefetching;
+        this.maxInflightReceiveBatches = builder.maxInflightReceiveBatches;
+        this.maxDoneReceiveBatches = builder.maxDoneReceiveBatches;
+    }
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    /**
+     * @return the optional maximum number of messages that are batched together in a single request.
+     */
+    public Integer maxBatchItems() {
+        return maxBatchItems;
+    }
+
+    /**
+     * @return the optional maximum number of batchKeys to keep track of.
+     */
+    public Integer maxBatchKeys() {
+        return maxBatchKeys;
+    }
+
+    /**
+     * @return the maximum number of items to allow to be buffered for each batchKey.
+     */
+    public Integer maxBufferSize() {
+        return maxBufferSize;
+    }
+
+    public Integer maxDoneReceiveBatches() {
+        return maxDoneReceiveBatches;
+    }
+
+    /**
+     * @return the optional maximum amount of time that an outgoing call waits to be batched with messages of the same type.
+     */
+    public Duration maxBatchOpenDuration() {
+        return maxBatchOpenDuration;
+    }
```

Comment:
The names of these methods don't seem to have much of a pattern to them. They're all over the place. Some places we call things "time" or "duration". Some places we're talking about batches, sometimes keys, and sometimes there's no noun. Is there a way to structure these settings in their names?

It would be good to use naming to try to make it clear what settings are related to each other. E.g. which are for request vs response? Which are related to limiting batch sizes?

Reply:
These names were directly taken from V1 . We can discuss this